Src/WitsmlExplorer.Frontend/components/ContextMenus/CopyToServer.tsx

```diff
@@ -31,38 +33,63 @@ export const onClickCopyToServer = async (
   dispatchOperation: DispatchOperation
 ) => {
   dispatchOperation({ type: OperationType.HideContextMenu });
-  const wellUid = toCopy[0].wellUid;
-  const wellboreUid = toCopy[0].wellboreUid;
-  const wellName = toCopy[0].wellName;
-  const wellboreName = toCopy[0].wellboreName;
+  const sourceWellUid = toCopy[0].wellUid;
+  const sourceWellboreUid = toCopy[0].wellboreUid;
+  const sourceWellName = toCopy[0].wellName;
+  const sourceWellboreName = toCopy[0].wellboreName;
 
-  const wellboreRef = {
-    wellUid: wellUid,
-    wellboreUid: wellboreUid,
-    wellName: wellName,
-    wellboreName: wellboreName
+  let targetWellboreRef: WellboreReference;
+
+  const dbQuery: UidMappingDbQuery = {
+    sourceServerId: sourceServer.id,
+    sourceWellId: sourceWellUid,
+    sourceWellboreId: sourceWellboreUid,
+    targetServerId: targetServer.id
   };
 
+  const mappings = await UidMappingService.queryUidMapping(dbQuery);
```

Comment:
Can't getTargetWellboreID also be used here? Not tested, but something like

```
const sourceWellboreName = toCopy[0].wellboreName;

  const { targetWellId, targetWellboreId } = await getTargetWellboreID({
    sourceServerId: sourceServer.id,
    sourceWellId: sourceWellUid,
    sourceWellboreId: sourceWellboreUid,
    targetServerId: targetServer.id
  });

  let wellbore: Wellbore;
  try {
    wellbore = await WellboreService.getWellbore(
      targetWellId,
      targetWellboreId,
      null,
      targetServer
    );
  } catch {
    return; // Cancel the operation if unable to authorize to the target server.
  }

  const targetWellboreRef: WellboreReference = {
    wellUid: targetWellId,
    wellboreUid: targetWellboreId,
    wellName: wellbore?.wellName ?? sourceWellName,
    wellboreName: wellbore?.name ?? sourceWellboreName
  };
```

Reply:
Yes, I'll check that again. However, I guess this would probably need a little bit more refactoring.